
For Bug 214607: [RHEL5 / FC6] - update cifs_getattr to set preferred IO size

Thanks,

-Eric

---

GIT 5fe14c851efedf95b0e7652a3a7b93ec899d1599 git+ssh://master.kernel.org/pub/scm/linux/kernel/git/sfrench/cifs-2.6.git

commit 
Author: Steve French <sfrench@us.ibm.com>
Date:   Tue Nov 7 19:26:33 2006 +0000

    [CIFS] Explicitly set stat->blksize
    
    CIFS may perform I/O over the network in larger chunks than the page size,
    so it should explicitly set stat->blksize to ensure optimal I/O bandwidth
    
    Signed-off-by: Dave Kleikamp <shaggy@linux.vnet.ibm.com>
    Signed-off-by: Steve French <sfrench@us.ibm.com>

Index: linux-2.6.18-1.2732.el5/fs/cifs/inode.c
===================================================================
--- linux-2.6.18-1.2732.el5.orig/fs/cifs/inode.c
+++ linux-2.6.18-1.2732.el5/fs/cifs/inode.c
@@ -1080,8 +1080,10 @@ int cifs_getattr(struct vfsmount *mnt, s
 	struct kstat *stat)
 {
 	int err = cifs_revalidate(dentry);
-	if (!err)
+	if (!err) {
 		generic_fillattr(dentry->d_inode, stat);
+		stat->blksize = CIFS_MAX_MSGSIZE;
+	}
 	return err;
 }
 

